ion/src/main/java/com/fasterxml/jackson/dataformat/ion/IonParser.java

```diff
@@ -390,8 +390,25 @@ public int getIntValue() throws IOException {
     // @since 2.17
     private int _getIntValue() throws IOException {
         try {
-            return _reader.intValue();
-        } catch (IonException e) {
+            if (getNumberType() == NumberType.LONG) {
+                int result = _reader.intValue();
+                if ((long) result != _reader.longValue()) {
+                    this.reportOverflowInt();
+                }
+                return result;
+            }
+            if (getNumberType() == NumberType.BIG_INTEGER) {
+                BigInteger bigInteger = _reader.bigIntegerValue();
+                if (BI_MIN_INT.compareTo(bigInteger) > 0 || BI_MAX_INT.compareTo(bigInteger) < 0) {
+                    this.reportOverflowInt();
+                }
+                return bigInteger.intValue();
+            } else {
+                return _reader.intValue();
+            }
+        } catch (IonException
+                // 15-Jan-2024, tatu: other OSS-Fuzz tests suggest we need this:
```

Comment:
Let's not add this back (I think @tgregg is working on handling these edge cases)

Reply:
Correct. If fuzzing identifies any new leaked exceptions I'd like to fix those in ion-java; in the meantime we can cover them in `failing` tests.